Firebase/Database/Login/FAuthTokenProvider.m

```diff
@@ -15,9 +15,14 @@
  */
 
 #import "FAuthTokenProvider.h"
-#import "FUtilities.h"
+
+#import <FirebaseAuthInterop/FIRAuthInterop.h>
 #import <FirebaseCore/FIRAppInternal.h>
+#import <FirebaseCore/FIRComponent.h>
```

Comment:
Instead of this class using the component system directly, we could pass in the `auth` instance in the constructor instead of the `app`. This would simplify testing greatly too since we could just use fakes instead of having to mock `FIRApp` at all.

The notification can still be used since `Auth` passes in `self` as the object for the notification, so we can compare `notification.object == self.auth` instead of the app and the app from the notification's dictionary.

Reply:
Done.